src/AppInstallerCLIE2ETests/PowerShell/PowerShellModule.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace AppInstallerCLIE2ETests.PowerShell
+{
+    using NUnit.Framework;
+    using System;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading;
+
+    /// <summary>
+    /// Basic E2E smoke tests for verifying the behavior of the PowerShell module cmdlets.
+    /// </summary>
+    [Category("PowerShell")]
+    public class PowerShellModule
+    {
+        // TODO: Consider using Pester framework for conducting more extensive PowerShell module tests.
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            TestCommon.RunAICLICommand("source add", $"-n {Constants.TestSourceName} {Constants.TestSourceUrl}");
+        }
+
+        [OneTimeTearDown]
+        public void TearDown()
+        {
+            // TODO: This is a workaround to an issue where the server takes longer than expected to terminate when
+            // running from the E2E tests. This can cause other E2E tests to fail when attempting to reset the test source.
+            if (IsRunning(Constants.WinGetServerExeName))
+            {
+                // There should only be one WinGetServer process running at a time.
+                Process serverProcess = Process.GetProcessesByName(Constants.WinGetServerExeName).First();
+                serverProcess.Kill();
+            }
+
+            TestCommon.RunAICLICommand("source remove", $"{Constants.TestSourceName}");
+        }
+
+        [Test]
+        public void AssertServerShutdownAfterExecution()
+        {
+            if (!Environment.Is64BitProcess)
+            {
+                return;
+            }
+
+            TestCommon.RunPowerShellCommandWithResult(Constants.GetSourceCmdlet, $"-Name {Constants.TestSourceName}");
+
+            Assert.IsTrue(IsRunning(Constants.WinGetServerExeName), $"{Constants.WinGetServerExeName} is not running.");
+            Process serverProcess = Process.GetProcessesByName(Constants.WinGetServerExeName).First();
+
+            // Wait a maximum of 30 seconds for the server process to exit.
+            bool serverProcessExit = serverProcess.WaitForExit(30000);
+            Assert.IsTrue(serverProcessExit, $"{Constants.WinGetServerExeName} failed to terminate after creating COM object.");
+        }
+
+        [Test]
+        public void GetWinGetSource()
+        {
+            // Running the x86 PowerShell Module requires PowerShell Core (x86). The tests currently only target PowerShell Core (x64)

```

Comment:
>// Running the x86 PowerShell Module requires PowerShell Core (x86). The tests currently only target PowerShell Core (x64) [](http://example.com/codeflow?start=12&length=122)

nit: Since all tests target x64, this comment should move to around top of the test class.

And do we plan to target x86 in the future?

Reply:
To support x86 tests, we will need to manually install powershell core (x86) and have the E2E tests target that when running the tests. Since this is mainly just for smoke-testing, I don't currently have plans to target x86. Testing should be carried out through the Pester framework which will be more thorough since that is required for publishing to the gallery.